bug: Compiler error because of node type reference without node types

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.8.2

### Current Behavior

The compiler errors during a build because of a type import from `node:child_process`, which cannot be found. This seems to be not found because the stencil project is set up to not include node types.

### Expected Behavior

Do not error the build because of a type import from node in the internal compiler implementation when the stencil project itself does not have node types.

### System Info

```shell
      System: node 19.6.0
    Platform: windows (10.0.19045)
   CPU Model: Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz (4 cpus)
    Compiler: C:\stencil-bug\node_modules\@stencil\core\compiler\stencil.js
       Build: 1702322155
     Stencil: 4.8.2
  TypeScript: 5.2.2
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.26.0
```


### Steps to Reproduce

```sh
npm ci
# ...
npm run build
# ... ok
npm run build
# ...
[ ERROR ]  TypeScript: node_modules/@stencil/core/internal/stencil-private.d.ts:2:53
           Cannot find module 'child_process' or its corresponding type declarations.

      L1:  import { result } from './utils';
      L2:  import type { Serializable as CPSerializable } from 'child_process';
```

Happens when `lib` or `typeRoots` are set in `tsconfig.json` without node types

### Code Reproduction URL

https://github.com/silvertech-daniel/stencil-bug

### Additional Information

_No response_

--- a/scripts/bundles/internal.ts
+++ b/scripts/bundles/internal.ts
@@ -1,3 +1,4 @@
+import { generateDtsBundle } from 'dts-bundle-generator';
 import fs from 'fs-extra';
 import { join } from 'path';
 
@@ -55,6 +56,29 @@ async function copyStencilInternalDts(opts: BuildOptions, outputInternalDir: str
   const privateDtsSrcPath = join(declarationsInputDir, 'stencil-private.d.ts');
   const privateDtsDestPath = join(outputInternalDir, 'stencil-private.d.ts');
   let privateDts = cleanDts(await fs.readFile(privateDtsSrcPath, 'utf8'));
+
+  // @stencil/core/internal/child_process.d.ts
+  const childProcessSrcPath = join(declarationsInputDir, 'child_process.d.ts');
+  const childProcessDestPath = join(outputInternalDir, 'child_process.d.ts');
+
+  // we generate a tiny tiny bundle here of just
+  // `src/declarations/child_process.ts` so that `internal/stencil-private.d.ts`
+  // can import from `'./child_process'` without worrying about resolving the
+  // types from `node_modules`.
+  const childProcessDts = generateDtsBundle([
+    {
+      filePath: childProcessSrcPath,
+      libraries: {
+        // we need to mark this library so that types imported from it are inlined
+        inlinedLibraries: ['child_process'],
+      },
+      output: {
+        noBanner: true,
+        exportReferencedTypes: false,
+      },
+    },
+  ]).join('\n');
+  await fs.writeFile(childProcessDestPath, childProcessDts);
 
   // the private `.d.ts` imports the `Result` type from the `@utils` module, so
   // we need to rewrite the path so it imports from the right relative path

--- a/scripts/utils/bundle-dts.ts
+++ b/scripts/utils/bundle-dts.ts
@@ -1,4 +1,4 @@
-import { EntryPointConfig, generateDtsBundle, OutputOptions } from 'dts-bundle-generator/dist/bundle-generator.js';
+import { EntryPointConfig, generateDtsBundle, OutputOptions } from 'dts-bundle-generator';
 import fs from 'fs-extra';
 
 import { BuildOptions } from './options';

--- a/src/declarations/child_process.ts
+++ b/src/declarations/child_process.ts
@@ -1,0 +1,5 @@
+import type { Serializable } from 'child_process';
+
+// this is just so that we can bundle this type for `stencil-private.d.ts`
+// without having to run dts-bundle-generator on that whole file
+export { Serializable as CPSerializable };

--- a/src/declarations/stencil-private.ts
+++ b/src/declarations/stencil-private.ts
@@ -1,7 +1,7 @@
 import { result } from '@utils';
-import type { Serializable as CPSerializable } from 'child_process';
 
 import type { InMemoryFileSystem } from '../compiler/sys/in-memory-fs';
+import type { CPSerializable } from './child_process';
 import type {
   BuildEvents,
   BuildLog,

